IdentityCore/src/validation/MSIDAuthorityFactory.m

```diff
@@ -70,4 +71,17 @@ + (MSIDAuthority *)authorityFromUrl:(NSURL *)url
     return nil;
 }
 
++ (MSIDAuthority *)authorityFromJSONDictionary:(NSDictionary *)json
```

Comment:
we also need to account for a case when we don't get authority_url in JSON

Reply:
For this case, we will need to inject authority url manually into the json dictionary before parsing it here. It will be done on the broker side in sso request handler.